lib/concurrent/synchronization/abstract_struct.rb

```diff
@@ -6,9 +6,18 @@ module Synchronization
     module AbstractStruct
 
       # @!visibility private
-      def initialize(*values)
+      def initialize(*values, **kw_values)
         super()
-        ns_initialize(*values)
+        ns_initialize(*values, **kw_values)
+      end
+
+      # @!macro [attach] struct_keyword_init
+      #
+      #   Returns `true` if the struct uses keyword arguments.
+      #
+      #   @return [Boolean] true if struct uses keyword arguments
+      def keyword_init?
+        self.class::KEYWORD_INIT
       end
```

Comment:
The `keyword_init?` on struct instance probably has not usage other then the initialization, I think it can be omitted or moved to the struct class replacing the KEYWORD_INIT constant.